internal/pkg/agent/application/coordinator/coordinator_state.go

```diff
@@ -135,16 +135,25 @@ func (c *Coordinator) refreshState() {
 // Coordinator state and sets stateNeedsRefresh.
 // Must be called on the main Coordinator goroutine.
 func (c *Coordinator) applyComponentState(state runtime.ComponentComponentState) {
+
+	// check for any component updates to the known PID, so we can update the component monitoring
 	found := false
+	pidRequiresUpdate := false
 	for i, other := range c.state.Components {
 		if other.Component.ID == state.Component.ID {
+			if other.State.CheckinPid != state.State.CheckinPid {
+				pidRequiresUpdate = true
+			}
 			c.state.Components[i] = state
 			found = true
 			break
 		}
 	}
 	if !found {
 		c.state.Components = append(c.state.Components, state)
+		if state.State.CheckinPid != 0 {
```

Comment:
Does the component only send the PID once or only when it changes?

The reason I ask is it seems like to me at this point, every checkin would result in the 30 second ticker recomputing the component model.

Reply:
It should only mark `pidRequiresUpdate` if the component has an updated PID, or it's a new component with a non-zero pid. I don't see any part in the logic where `pidRequiresUpdate` would always be true?